packages/builder/src/components/common/bindings/BindingPanel.svelte

```diff
@@ -51,8 +51,9 @@
   })
   $: codeMirrorHints = bindings?.map(x => `$("${x.readableBinding}")`)
 
-  const updateValue = value => {
-    valid = isValid(readableToRuntimeBinding(bindings, value))
+  const updateValue = val => {
+    valid = isValid(readableToRuntimeBinding(bindings, val))
+    value = val
```

Comment:
Just wondering was this fixing an issue? Normally whenever we update the value here with the `change` event, the new `value` prop will be passed in from the parent component again.

This is updating the value even it's not valid, which I don't think will cause any issues... but just wondering what prompted the change.

Reply:
Yeah so the issue I was getting around here was some weirdness around clearing the state and then adding helpers and such, the `addHelper`/`addBinding` functions both depend on the value of `value` to be able to be up to date so that you don't end up in scenarios where some old value comes back, because the state is a bit out of sync with what is in the text area.

Around the a validity, I did wonder about this myself, since anything bound to the component could pull out an invalid expression.